pkg/pillar/cmd/nim/nim.go

```diff
@@ -1188,3 +1199,61 @@ func isAssigned(ctx *nimContext, ifname string) bool {
 	}
 	return false
 }
+
+const (
+	configDPCDir = types.IdentityDirname + "/DevicePortConfig"
+	runDPCDir    = "/run/global/DevicePortConfig"
+	maxReadSize  = 16384 // Punt on too large files
+)
+
+// ingestPortConfig reads all json files in configDPCDir, ensures they have
+// TimePriority, and adds a OriginFile to them and writes to runDPCDir
+func ingestDevicePortConfig(ctx *nimContext) {
+	locations, err := ioutil.ReadDir(configDPCDir)
+	if err != nil {
+		// Directory might not exist
+		return
+	}
+	for _, location := range locations {
+		if !location.IsDir() {
+			ingestDevicePortConfigFile(ctx, configDPCDir, runDPCDir,
+				location.Name())
+		}
+	}
+}
+
+func ingestDevicePortConfigFile(ctx *nimContext, oldDirname string, newDirname string, name string) {
+	filename := path.Join(oldDirname, name)
+	log.Noticef("ingestDevicePortConfigFile(%s)", filename)
+	str, _, err := fileutils.StatAndRead(log, filename, maxReadSize)
+	if err != nil {
+		log.Errorf("Failed to read file %s: %v", filename, err)
+		return
+	}
+	if str == "" {
+		log.Errorf("Ignore empty file %s", filename)
+		return
+	}
+
+	var dpc types.DevicePortConfig
+	err = json.Unmarshal([]byte(str), &dpc)
+	if err != nil {
+		log.Errorf("Could not unmarshall data in file %s: %s",
+			filename, err)
+		return
+	}
+	dpc.DoSanitize(log, true, false, "", true)
+	dpc.OriginFile = filename
+
+	// Save New config to file.
+	var data []byte
+	data, err = json.Marshal(dpc)
+	if err != nil {
+		log.Fatalf("Failed to marshall new dpc err %s", err)
+	}
+	filename = path.Join(newDirname, name)
+	err = fileutils.WriteRename(filename, data)
+	if err != nil {
+		log.Errorf("Failed to write new dpc to %s: %s", filename, err)
```

Comment:
And here I actually convinced that `dpc` should be rather "device port config", since this is a error message. Yeah, the filename would probably still give a user an idea what dpc is actually stands for. But still I am for the clarity

Reply:
Will fix.